packages/block-editor/src/components/block-navigation/block-select-button.js

```diff
@@ -0,0 +1,81 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import {
+	__experimentalGetBlockLabel as getBlockLabel,
+	getBlockType,
+} from '@wordpress/blocks';
+import { Button, VisuallyHidden } from '@wordpress/components';
+import { useInstanceId } from '@wordpress/compose';
+import { forwardRef } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import BlockIcon from '../block-icon';
+import { getBlockPositionDescription } from './utils';
+
+function BlockNavigationBlockSelectButton(
+	{
+		className,
+		block,
+		isSelected,
+		onClick,
+		position,
+		siblingCount,
+		level,
+		tabIndex,
+		onFocus,
+	},
+	ref
+) {
+	const { name, attributes } = block;
```

Comment:
You can use multi-level destructuring to simplify this a bit:
```suggestion
function BlockNavigationBlockSelectButton(
	{
		className,
		block: { name, attributes },
		isSelected,
		onClick,
		position,
		siblingCount,
		level,
		tabIndex,
		onFocus,
	},
	ref
) {
```

Reply:
Thanks for taking a look at the PR. 

My feeling is that even though it's not an actual coding standard for the project, destructuring like this is not commonly used across this codebase, and I'd prefer to stay consistent with other code.

Reply:
For what it's worth, it seems like this pattern is becoming [a bit more common lately](https://github.com/WordPress/gutenberg/pull/22499/files#diff-f5fc3d5a78f58a5274bc196ebc657c5bR98-R113).